crates/fluvio-cluster/src/start/common.rs

```diff
@@ -30,34 +37,33 @@ pub async fn try_connect_to_sc(
     async fn try_connect_sc(
         fluvio_config: &FluvioConfig,
         expected_version: &Version,
-    ) -> Option<Fluvio> {
+        timeout: Duration,
+    ) -> Result<Fluvio, TryConnectError> {
         use tokio::select;
 
         select! {
-            _ = &mut sleep(Duration::from_secs(10)) => {
+            _ = &mut sleep(timeout) => {
                 debug!("timer expired");
-                None
+                Err(TryConnectError::Timeout)
             },
 
             connection = Fluvio::connect_with_config(fluvio_config) =>  {
-
                 match connection {
                     Ok(fluvio) => {
                         let current_version = fluvio.platform_version();
                         if current_version == expected_version {
                             debug!("Got updated SC Version{}", &expected_version);
-                            Some(fluvio)
+                            Ok(fluvio)
                         } else {
                             warn!("Current Version {} is not same as expected: {}",current_version,expected_version);
-                            None
+                            Err(TryConnectError::UnexpectedVersion)
```

Comment:
can you add that information in the warning to `TryConnectError::UnexpectedVersion`